pkg/types/aws/machinepool.go

```diff
@@ -24,6 +24,11 @@ type MachinePool struct {
 	//
 	// +optional
 	EC2RootVolume `json:"rootVolume"`
+
+	// The name of a user-created IAM Role to use. This takes the place of an installer-created role.
```

Comment:
The description should start with the name of the field. And some other suggestions on this description to keep it consistent with other BYO fields.
```suggestion
	// IAMRole is the name of the IAM Role to use for the instance profile of the machine. Leave unset to have the installer create the IAM Role on your behalf.
```